### ISSUE TITLE
feature ordering gets scrambled when written to .pmml file and reloaded with pypmml

### ISSUE THREAD

**Issue by doolingdavid** [jpmml/sklearn2pmml#238]
Hi, Thank you for the fantastic sklearn2pmml. I am new to using it and have been able to generate a pmml file capturing a gradient boosting regressor model:
```python
from sklearn_pandas import DataFrameMapper
from sklearn2pmml.decoration import CategoricalDomain, ContinuousDomain

mapper = DataFrameMapper(
    [([column], ContinuousDomain()) for column in inputcols])

estimator = GradientBoostingRegressor(max_depth=8,
            n_estimators=350,learning_rate=.05,random_state=2,
                   min_samples_leaf=5)

xdomainscaler = StandardScaler()

pipeline_0 = PMMLPipeline([
    ("mapper", mapper),
    ("scaler", xdomainscaler),
    ("estimator", estimator)
])

dg_X = dg[inputcols]
dg_y0 = dg[outputcols[0]]

print(dg_X.head(2))
        I1   I2   I3    I4   I5   I6    I7
8946   3.0  2.0  9.0  16.0  2.4  1.0  3.75
11901  2.0  8.0  9.0   8.0  2.4  4.0  3.75

pipeline_0.fit(dg_X, dg_y0)
sklearn2pmml(pipeline_0, "pipeline_0.pmml")
```
For some reason the model imported from the saved pmml file has the input columns scrambled out of order, requiring that a user reorder their input to match. Is this a problem with how i am saving the pipeline, or is there anyway to ensure that the order of the input columns is preserved from the orignal pipeline to the pmml file?
```python
from pypmml import Model
model = Model.fromFile("pipeline_0.pmml")

model.inputNames
['I5', 'I2', 'I6', 'I7', 'I4', 'I3', 'I1']
```

**Comment by vruusmann** [comment]
> For some reason the model imported from the saved pmml file has the input columns scrambled out of order

In PMML fields are identified by name, not by position. The closest Python data structure to this is `dict` (keys - field names, values - field values).

In the light of the above, there is no "correct order" of fields.

> requiring that a user reorder their input to match

If the PyPMML library chooses to allow unnamed input (eg. Numpy matrix instead of a Pandas `DataFrame`), then it's PyPMML's task to identify/establish the order of fields.

In my opinion, the positional ordering/identification of fields is a very brittle, and should be avoided.

> Is this a problem with how i am saving the pipeline

The `DataDictionary/DataField` elements are definitely ordered after their `DataFrameMapper` declaration order. The `<Mode>/MiningSchema/MiningField` elements are ordered after their actual usages - whatever field is first needed by the model first (eg. the top-level split in the decision tree model) appears in the first position etc.

> to ensure that the order of the input columns is preserved from the orignal pipeline to the pmml file?

There's always a possibility that the PyPMML library is reordering field declarations all by itself.

**Comment by vruusmann** [comment]
Resolution - won't fix, everything works as intended.
